infrasim/ipmiconsole/common.py

```diff
@@ -85,18 +86,22 @@ def init_env(instance):
         raise IpmiError(
             "Warning: there is no node {} workspace. "
             "Please start node {} first.".format(instance, instance))
-    try:
-        with open("{}/{}/.{}-bmc.pid".format(
-                config.infrasim_home, instance, instance), "r") as f:
-            pid = f.readline().strip()
-            if not os.path.exists("/proc/{}".format(pid)):
-                raise Exception
-    except Exception:
-        logger.error("Warning: node {} has not started BMC. "
-                     "Please start node {} first.".format(instance, instance))
-        raise IpmiError(
-            "Warning: node {} has not started BMC. "
-            "Please start node {} first.".format(instance, instance))
+
+    PS_QEMU = "ps ax | grep qemu"
```

Comment:
check content ".<instance>-node.pid" is reasonable.
call shell command is expensive.